In [1]:
import pandas as pd 
import numpy as np

/Users/titooyewole/.pyenv/versions/3.9.6/lib/python3.9/site-packages/pandas/compat/__init__.py:124: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


In [88]:
df = pd.read_csv("/Users/titooyewole/Downloads/python-portfolio-project-starter-files/insurance.csv")
df.describe()




age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [81]:
bmi = df[(df["bmi"] > 30.66)]
bmi.describe()


64

In [90]:
bmi_male = bmi[(bmi["sex"] == "male")]
bmi_male.describe()


age         bmi    children       charges
count  340.000000  340.000000  340.000000    340.000000
mean    40.561765   35.805015    1.105882  17098.134487
std     14.299752    4.091294    1.172683  15382.357243
min     18.000000   30.685000    0.000000   1135.940700
25%     28.000000   32.700000    0.000000   5413.131963
50%     41.000000   34.930000    1.000000  10597.387000
75%     53.000000   37.736250    2.000000  33876.232187
max     64.000000   53.130000    5.000000  60021.398970

In [78]:
bmi_female = bmi[(bmi["sex"] == "female")]
bmi_female.describe()

age         bmi    children       charges
count  306.000000  306.000000  306.000000    306.000000
mean    40.986928   35.658578    1.120915  14361.403871
std     14.331662    3.794686    1.201810  13699.534011
min     18.000000   30.685000    0.000000   1621.882700
25%     28.250000   32.680000    0.000000   5326.088313
50%     42.500000   34.800000    1.000000   9833.596525
75%     53.000000   38.045000    2.000000  14294.152725
max     64.000000   48.070000    5.000000  63770.428010

In [91]:
age = df[(df["age"] > 39.207025)]
age.describe()

age         bmi    children       charges
count  664.000000  664.000000  664.000000    664.000000
mean    51.548193   31.260715    1.126506  16430.512562
std      7.037548    5.892397    1.181413  11754.526040
min     40.000000   18.050000    0.000000   5415.661200
25%     46.000000   26.698750    0.000000   9033.060162
50%     51.000000   31.160000    1.000000  11657.916975
75%     57.000000   35.727500    2.000000  19839.727488
max     64.000000   49.060000    5.000000  63770.428010

In [94]:
age_male =  age[(age["sex"]) == "male"]
age_male.describe()


age         bmi    children       charges
count  330.000000  330.000000  330.000000    330.000000
mean    51.460606   31.429348    1.121212  17071.246085
std      7.052126    5.915307    1.196692  12767.755398
min     40.000000   18.335000    0.000000   5415.661200
25%     45.250000   26.638750    0.000000   8805.747225
50%     51.000000   31.350000    1.000000  11359.707950
75%     57.000000   35.942500    2.000000  21662.594787
max     64.000000   49.060000    5.000000  62592.873090

In [95]:
age_female =  age[(age["sex"]) == "female"]
age_female.describe()

age         bmi    children       charges
count  334.000000  334.000000  334.000000    334.000000
mean    51.634731   31.094102    1.131737  15797.452495
std      7.032622    5.873784    1.167893  10640.806259
min     40.000000   18.050000    0.000000   5910.944000
25%     46.000000   26.821250    0.000000   9411.983750
50%     51.000000   30.887500    1.000000  11845.382375
75%     57.750000   35.200000    2.000000  16009.018995
max     64.000000   48.070000    5.000000  63770.428010

In [84]:
smoker = df[(df["smoker"] == "yes")]
smoker.describe()


age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
11     62  female  26.290         0    yes  southeast  27808.72510
14     27    male  42.130         0    yes  southeast  39611.75770
19     30    male  35.300         0    yes  southwest  36837.46700
23     34  female  31.920         1    yes  northeast  37701.87680
...   ...     ...     ...       ...    ...        ...          ...
1313   19  female  34.700         2    yes  southwest  36397.57600
1314   30  female  23.655         3    yes  northwest  18765.87545
1321   62    male  26.695         0    yes  northeast  28101.33305
1323   42  female  40.370         2    yes  southeast  43896.37630
1337   61  female  29.070         0    yes  northwest  29141.36030

[274 rows x 7 columns]

In [60]:
female = df[(df["sex"] == "female")]
female.describe()


age         bmi    children       charges
count  662.000000  662.000000  662.000000    662.000000
mean    39.503021   30.377749    1.074018  12569.578844
std     14.054223    6.046023    1.192115  11128.703801
min     18.000000   16.815000    0.000000   1607.510100
25%     27.000000   26.125000    0.000000   4885.158700
50%     40.000000   30.107500    1.000000   9412.962500
75%     51.750000   34.313750    2.000000  14454.691825
max     64.000000   48.070000    5.000000  63770.428010

In [64]:
female_smoker = female[(female["smoker"] == "yes")]
female_smoker.describe()

age         bmi    children       charges
count  115.000000  115.000000  115.000000    115.000000
mean    38.608696   29.608261    1.008696  30678.996276
std     14.217419    6.663802    1.135516  11907.536381
min     18.000000   17.195000    0.000000  13844.506000
25%     26.500000   24.650000    0.000000  19696.432100
50%     39.000000   28.380000    1.000000  28950.469200
75%     48.500000   34.352500    2.000000  40918.314500
max     64.000000   47.600000    5.000000  63770.428010

In [61]:
male = df[(df["sex"] == "male")]
male.describe()


age         bmi    children       charges
count  676.000000  676.000000  676.000000    676.000000
mean    38.917160   30.943129    1.115385  13956.751178
std     14.050141    6.140435    1.218986  12971.025915
min     18.000000   15.960000    0.000000   1121.873900
25%     26.000000   26.410000    0.000000   4619.134000
50%     39.000000   30.687500    1.000000   9369.615750
75%     51.000000   34.992500    2.000000  18989.590250
max     64.000000   53.130000    5.000000  62592.873090

In [66]:
male_smoker = male[(male["smoker"] == "yes")]
male_smoker.describe()

age         bmi    children       charges
count  159.000000  159.000000  159.000000    159.000000
mean    38.446541   31.504182    1.188679  33042.005975
std     13.751247    5.952028    1.170144  11202.670862
min     18.000000   17.290000    0.000000  12829.455100
25%     27.000000   27.360000    0.000000  21241.526875
50%     38.000000   31.130000    1.000000  36085.219000
75%     49.500000   35.565000    2.000000  41797.589050
max     64.000000   52.580000    4.000000  62592.873090

In [97]:
kids = df[(df["children"]) > 1.094918]
kids.describe()

age         bmi    children       charges
count  440.000000  440.000000  440.000000    440.000000
mean    40.020455   30.840580    2.593182  14847.398459
std     11.831083    6.181806    0.775742  12301.152463
min     18.000000   16.815000    2.000000   2304.002200
25%     30.000000   26.210000    2.000000   6405.380863
50%     40.000000   30.790000    2.000000  10070.109500
75%     49.000000   34.777500    3.000000  19053.589130
max     64.000000   48.070000    5.000000  60021.398970

In [98]:
men_with_kids = kids[(kids["sex"]) == "male"]
men_with_kids.describe()

age         bmi    children       charges
count  225.000000  225.000000  225.000000    225.000000
mean    39.031111   31.033711    2.613333  15884.624438
std     11.361900    6.028055    0.794400  13640.401142
min     18.000000   16.815000    2.000000   2304.002200
25%     30.000000   26.400000    2.000000   6203.901750
50%     39.000000   30.800000    2.000000   9304.701900
75%     47.000000   35.200000    3.000000  22493.659640
max     64.000000   46.530000    5.000000  60021.398970

In [99]:
women_with_kids = kids[(kids["sex"]) == "female"]
women_with_kids.describe()

age         bmi    children       charges
count  215.000000  215.000000  215.000000    215.000000
mean    41.055814   30.638465    2.572093  13761.929412
std     12.243887    6.346499    0.757000  10646.356282
min     18.000000   17.195000    2.000000   2801.258800
25%     31.000000   26.035000    2.000000   6628.987650
50%     41.000000   30.685000    2.000000  10381.478700
75%     51.500000   34.490000    3.000000  15948.953240
max     64.000000   48.070000    5.000000  47305.305000

average cost of insurance is $13270.42
average age is 39
average number of children is 1
average bmi is 30.66
20.47% of the population smoke
The average cost of insurance for a smoker is $32050.23
41.97% of smokers are females and the average cost of insurance for a female smoker is $30678.99
58.03% of smokers are men and the average cost of insurance for a male smoker is $33042.00
It is impotant to note that the bmi of smokers is slightly above the average bmi of the entire population.
48.28% of the population has a bmi above average.
52.63% of the population with bmi above average is made up of men
47.37% of the population with bmi above average is made up of women
49.63% of the population are older than the average age
49.70% of the older population are men
50.30% of the older population are women
32.88% of the population have more kids than the average amount
51.14% of the 32.88% are men
48.86% of the 32.88% are women
We will assume that a bmi above average is an indicator of an unhealthy lifestyle
Men typically pay more on insurance than women. This could be because
i. They makeup most of the population (50.52%)
ii. They make up more of the smokers population(58.03%)
iii. They have a higher bmi than females
iv. They have more children than females

These are all speculations


In [55]:
sw_reg = df[(df["region"]== "southwest")]
sw_reg.describe()



age         bmi    children       charges
count  325.000000  325.000000  325.000000    325.000000
mean    39.455385   30.596615    1.141538  12346.937377
std     13.959886    5.691836    1.275952  11557.179101
min     19.000000   17.400000    0.000000   1241.565000
25%     27.000000   26.900000    0.000000   4751.070000
50%     39.000000   30.300000    1.000000   8798.593000
75%     51.000000   34.600000    2.000000  13462.520000
max     64.000000   47.600000    5.000000  52590.829390

In [89]:
sw_age = sw_reg[(sw_reg["age"] > 39.207025 )]
sw_age.describe()

age         bmi    children       charges
count  162.000000  162.000000  162.000000    162.000000
mean    51.679012   31.387654    1.098765  14174.756317
std      6.952951    5.414712    1.170071  10281.739610
min     40.000000   20.100000    0.000000   5910.944000
25%     46.000000   27.500000    0.000000   8542.590000
50%     51.500000   30.800000    1.000000  10883.084500
75%     57.000000   35.775000    2.000000  13349.092500
max     64.000000   46.700000    5.000000  52590.829390

In [70]:
sw_smoker = sw_reg[(sw_reg["smoker"] == "yes")]
sw_smoker.describe() 

age        bmi   children       charges
count  58.000000  58.000000  58.000000     58.000000
mean   36.103448  31.005172   1.206897  32269.063494
std    13.739033   5.420831   1.280880  11270.279593
min    19.000000  18.300000   0.000000  13844.506000
25%    25.000000  27.625000   0.000000  20396.042022
50%    33.500000  31.350000   1.000000  35165.256500
75%    45.500000  34.775000   2.000000  39809.292750
max    64.000000  47.600000   5.000000  52590.829390

In [67]:
sw_male = sw_reg[(sw_reg["sex"] == "male")]
sw_male.describe()

age         bmi    children       charges
count  163.000000  163.000000  163.000000    163.000000
mean    39.208589   31.129448    1.159509  13412.883576
std     13.952501    5.586314    1.295416  12499.550025
min     19.000000   19.300000    0.000000   1241.565000
25%     26.500000   27.450000    0.000000   4658.699500
50%     39.000000   30.800000    1.000000   9391.346000
75%     51.000000   35.300000    2.000000  17119.976000
max     64.000000   45.900000    5.000000  52590.829390

In [71]:
sw_male_smoker = sw_male[(sw_male["smoker"] == "yes")]
sw_male_smoker.describe()

age        bmi   children       charges
count  37.000000  37.000000  37.000000     37.000000
mean   35.567568  31.502703   1.297297  32598.862854
std    12.431011   4.704046   1.288084  10782.406370
min    19.000000  19.300000   0.000000  15820.699000
25%    26.000000  27.800000   0.000000  22144.032000
50%    33.000000  32.700000   1.000000  35585.576000
75%    44.000000  35.300000   2.000000  39241.442000
max    61.000000  39.900000   4.000000  52590.829390

In [68]:
sw_female = sw_reg[(sw_reg["sex"] == "female")]
sw_female.describe()

age         bmi    children       charges
count  162.000000  162.000000  162.000000    162.000000
mean    39.703704   30.060494    1.123457  11274.411264
std     14.006161    5.763597    1.259821  10453.660658
min     19.000000   17.400000    0.000000   1727.785000
25%     27.000000   25.900000    0.000000   4895.811250
50%     40.000000   29.750000    1.000000   8530.837000
75%     51.750000   33.650000    2.000000  12642.990500
max     64.000000   47.600000    5.000000  48824.450000

In [100]:
sw_bmi = sw_reg[(sw_reg["bmi"] > 30.663397)]
sw_bmi.describe()

age         bmi    children       charges
count  155.000000  155.000000  155.000000    155.000000
mean    41.683871   35.384516    1.193548  15233.914251
std     14.157167    3.512448    1.238436  14160.770762
min     19.000000   30.800000    0.000000   1261.442000
25%     30.000000   32.800000    0.000000   5568.988500
50%     41.000000   34.800000    1.000000   9872.701000
75%     54.000000   37.100000    2.000000  14338.697685
max     64.000000   47.600000    5.000000  52590.829390

In [101]:
sw_kids = sw_reg[(sw_reg["children"] > 1.094918)]
sw_kids.describe()

age         bmi    children       charges
count  109.000000  109.000000  109.000000    109.000000
mean    39.752294   31.601835    2.688073  14252.615901
std     11.857297    5.484355    0.889207  12250.149406
min     19.000000   18.300000    2.000000   2913.569000
25%     30.000000   27.500000    2.000000   6059.173000
50%     39.000000   30.900000    2.000000   9414.920000
75%     49.000000   34.800000    3.000000  17942.106000
max     64.000000   47.600000    5.000000  47305.305000

The average cost of insurance in this region is $12346.937377
24.29% of the population live in the southwest region
17.84% of this region smoke
49.85% of southwest region are older than the average age
47.69% of this region are above the average bmi
33.54% of this region have more kids than the average

50.15% of the population are men
49.85% of the population are female

In [56]:
se_reg = df[(df["region"]== "southeast")]
se_reg.describe()


age         bmi    children       charges
count  364.000000  364.000000  364.000000    364.000000
mean    38.939560   33.355989    1.049451  14735.411438
std     14.164585    6.477648    1.177276  13971.098589
min     18.000000   19.800000    0.000000   1121.873900
25%     26.750000   28.572500    0.000000   4440.886200
50%     39.000000   33.330000    1.000000   9294.131950
75%     51.000000   37.812500    2.000000  19526.286900
max     64.000000   53.130000    5.000000  63770.428010

In [62]:
se_reg_women = se_reg[(se_reg["sex"] == "female")]
se_reg_women.describe()

age         bmi    children       charges
count  175.000000  175.000000  175.000000    175.000000
mean    39.108571   32.671257    1.051429  13499.669243
std     14.180069    6.083665    1.199945  12587.261506
min     18.000000   20.790000    0.000000   1607.510100
25%     27.000000   27.995000    0.000000   4382.836600
50%     39.000000   32.230000    1.000000   8582.302300
75%     51.000000   37.565000    2.000000  18220.806295
max     64.000000   47.410000    5.000000  63770.428010

In [63]:
se_reg_men = se_reg[(se_reg["sex"] == "male")]
se_reg_men.describe()

age         bmi    children       charges
count  189.000000  189.000000  189.000000    189.000000
mean    38.783069   33.990000    1.047619  15879.617173
std     14.186093    6.776649    1.159079  15083.360816
min     18.000000   19.800000    0.000000   1121.873900
25%     26.000000   29.700000    0.000000   4463.205100
50%     39.000000   34.100000    1.000000   9504.310300
75%     51.000000   38.060000    2.000000  21978.676900
max     64.000000   53.130000    5.000000  62592.873090

In [102]:
se_smoker = se_reg[(se_reg["smoker"] == "yes")]
se_smoker.describe()

age        bmi   children       charges
count  91.000000  91.000000  91.000000     91.000000
mean   39.736264  33.096703   1.000000  34844.996824
std    14.070327   7.017828   1.064581  11324.766536
min    18.000000  19.800000   0.000000  16577.779500
25%    27.000000  27.170000   0.000000  23155.105450
50%    42.000000  33.110000   1.000000  37484.449300
75%    51.000000  38.060000   2.000000  43391.859400
max    64.000000  52.580000   3.000000  63770.428010

In [103]:
se_bmi = se_reg[(se_reg["bmi"] > 30.663397)]
se_bmi.describe() 

age         bmi    children       charges
count  235.000000  235.000000  235.000000    235.000000
mean    39.148936   37.133191    1.034043  16296.077977
std     14.682362    4.513664    1.176202  15676.031084
min     18.000000   30.690000    0.000000   1135.940700
25%     25.000000   33.660000    0.000000   4589.027300
50%     39.000000   36.300000    1.000000   9504.310300
75%     52.000000   39.710000    2.000000  27535.165410
max     64.000000   53.130000    5.000000  63770.428010

In [104]:
se_kids = se_reg[(se_reg["children"] > 1.094918)]
se_kids.describe()

age         bmi    children       charges
count  112.000000  112.000000  112.000000    112.000000
mean    40.901786   33.294643    2.562500  16221.173578
std     12.135095    6.606029    0.814079  13736.785651
min     18.000000   20.130000    2.000000   2304.002200
25%     31.750000   28.490000    2.000000   6331.996450
50%     42.000000   33.330000    2.000000  10470.985200
75%     50.000000   37.647500    3.000000  19927.786950
max     64.000000   46.750000    5.000000  49577.662400

In [105]:
se_age = se_reg[(se_reg["age"] > 39.207025 )]
se_age.describe()

age         bmi    children       charges
count  180.000000  180.000000  180.000000    180.000000
mean    51.372222   33.295778    1.155556  18854.735752
std      7.204926    6.636242    1.213620  14085.292815
min     40.000000   19.800000    0.000000   5415.661200
25%     45.000000   27.912500    0.000000   8804.704150
50%     51.000000   33.495000    1.000000  12375.653900
75%     57.250000   38.060000    2.000000  24551.673195
max     64.000000   49.060000    5.000000  63770.428010

In [106]:
se_male = se_reg[(se_reg["sex"] == "male")]
se_male.describe()

age         bmi    children       charges
count  189.000000  189.000000  189.000000    189.000000
mean    38.783069   33.990000    1.047619  15879.617173
std     14.186093    6.776649    1.159079  15083.360816
min     18.000000   19.800000    0.000000   1121.873900
25%     26.000000   29.700000    0.000000   4463.205100
50%     39.000000   34.100000    1.000000   9504.310300
75%     51.000000   38.060000    2.000000  21978.676900
max     64.000000   53.130000    5.000000  62592.873090

In [107]:
se_female = se_reg[(se_reg["sex"] == "female")]
se_female.describe()

age         bmi    children       charges
count  175.000000  175.000000  175.000000    175.000000
mean    39.108571   32.671257    1.051429  13499.669243
std     14.180069    6.083665    1.199945  12587.261506
min     18.000000   20.790000    0.000000   1607.510100
25%     27.000000   27.995000    0.000000   4382.836600
50%     39.000000   32.230000    1.000000   8582.302300
75%     51.000000   37.565000    2.000000  18220.806295
max     64.000000   47.410000    5.000000  63770.428010


The region with the most amount of people is southeast. It makes up 27.2% of the population
The average cost of insurance in this region is $14735.41. This could be because:

25% of this region smokes
64.56% of this region are above the average bmi 
30.77% of this region have more than the average amount of children
49.45% of this population are older than the average age

51.92% are men
48.08 are women


In [108]:
ne_reg = df[(df["region"]== "northeast")]
ne_reg.describe()



age         bmi    children       charges
count  324.000000  324.000000  324.000000    324.000000
mean    39.268519   29.173503    1.046296  13406.384516
std     14.069007    5.937513    1.198949  11255.803066
min     18.000000   15.960000    0.000000   1694.796400
25%     27.000000   24.866250    0.000000   5194.322288
50%     39.500000   28.880000    1.000000  10057.652025
75%     51.000000   32.893750    2.000000  16687.364100
max     64.000000   48.070000    5.000000  58571.074480

In [115]:
ne_age = ne_reg[(ne_reg["age"] > 39.207025 )]
ne_age.describe()

age         bmi    children       charges
count  162.000000  162.000000  162.000000    162.000000
mean    51.518519   30.177160    1.043210  16283.087568
std      6.960601    5.687352    1.143976  10614.270344
min     40.000000   18.335000    0.000000   6600.205950
25%     46.000000   25.460000    0.000000   9510.853675
50%     51.000000   30.780000    1.000000  11934.690600
75%     57.000000   33.416250    2.000000  19747.243325
max     64.000000   48.070000    5.000000  48549.178350

In [109]:
ne_reg_women = ne_reg[(ne_reg["sex"] == "female")]
ne_reg_women.describe()

age         bmi    children       charges
count  161.000000  161.000000  161.000000    161.000000
mean    39.639752   29.324317    1.006211  12953.203151
std     14.014079    6.338764    1.109600  10591.929980
min     18.000000   16.815000    0.000000   2196.473200
25%     27.000000   24.225000    0.000000   5458.046450
50%     40.000000   29.165000    1.000000  10197.772200
75%     52.000000   33.250000    2.000000  15359.104500
max     64.000000   48.070000    4.000000  58571.074480

In [110]:
ne_reg_men = ne_reg[(ne_reg["sex"] == "male")]
ne_reg_men.describe()

age         bmi    children       charges
count  163.000000  163.000000  163.000000    163.000000
mean    38.901840   29.024540    1.085890  13854.005374
std     14.156626    5.528263    1.283315  11890.942381
min     18.000000   15.960000    0.000000   1694.796400
25%     26.000000   25.317500    0.000000   4880.990000
50%     39.000000   28.880000    1.000000   9957.721600
75%     51.000000   32.300000    2.000000  18276.474375
max     64.000000   42.655000    5.000000  48549.178350

In [111]:
ne_smoker = ne_reg[(ne_reg["smoker"] == "yes")]
ne_smoker.describe()

age        bmi   children       charges
count  67.000000  67.000000  67.000000     67.000000
mean   38.238806  28.565224   1.044776  29673.536473
std    13.348578   6.208727   1.065049  11589.559770
min    18.000000  17.195000   0.000000  12829.455100
25%    26.500000  23.607500   0.000000  19881.400425
50%    39.000000  28.500000   1.000000  28101.333050
75%    49.500000  32.537500   2.000000  39576.950850
max    63.000000  42.750000   3.000000  58571.074480

In [113]:
ne_bmi = ne_reg[(ne_reg["bmi"] > 30.663397)]
ne_bmi.describe() 

age         bmi    children       charges
count  131.000000  131.000000  131.000000    131.000000
mean    42.847328   34.967252    1.022901  17419.189469
std     13.873868    3.537103    1.139944  14077.041087
min     18.000000   30.685000    0.000000   1984.453300
25%     32.000000   32.252500    0.000000   7645.095600
50%     46.000000   33.725000    1.000000  11879.104050
75%     54.000000   37.050000    2.000000  25124.189550
max     64.000000   48.070000    4.000000  58571.074480

In [114]:
ne_kids = ne_reg[(ne_reg["children"] > 1.094918)]
ne_kids.describe()

age        bmi    children       charges
count  100.000000  100.00000  100.000000    100.000000
mean    39.130000   28.46200    2.620000  13786.926796
std     11.637327    6.39440    0.749141  10554.207026
min     18.000000   16.81500    2.000000   3279.868550
25%     29.750000   23.53625    2.000000   6748.129025
50%     39.000000   28.35750    2.000000   9602.620550
75%     48.000000   32.30000    3.000000  17632.463690
max     64.000000   48.07000    5.000000  48549.178350

The average cost of insurance in this region is $13406.384516
Northeast region makes up 24.22% of the population
49.69% are women
50.31% are men
20.62% smoke
40.31% are above the average bmi
30.86% have more than the average amount of kids
50% are above the average age


In [58]:
nw_reg = df[(df["region"]== "southwest")]
nw_reg.describe()

age         bmi    children       charges
count  325.000000  325.000000  325.000000    325.000000
mean    39.455385   30.596615    1.141538  12346.937377
std     13.959886    5.691836    1.275952  11557.179101
min     19.000000   17.400000    0.000000   1241.565000
25%     27.000000   26.900000    0.000000   4751.070000
50%     39.000000   30.300000    1.000000   8798.593000
75%     51.000000   34.600000    2.000000  13462.520000
max     64.000000   47.600000    5.000000  52590.829390

In [116]:
nw_age = nw_reg[(nw_reg["age"] > 39.207025 )]
nw_age.describe()

age         bmi    children       charges
count  162.000000  162.000000  162.000000    162.000000
mean    51.679012   31.387654    1.098765  14174.756317
std      6.952951    5.414712    1.170071  10281.739610
min     40.000000   20.100000    0.000000   5910.944000
25%     46.000000   27.500000    0.000000   8542.590000
50%     51.500000   30.800000    1.000000  10883.084500
75%     57.000000   35.775000    2.000000  13349.092500
max     64.000000   46.700000    5.000000  52590.829390

In [117]:
nw_reg_women = nw_reg[(nw_reg["sex"] == "female")]
nw_reg_women.describe()

age         bmi    children       charges
count  162.000000  162.000000  162.000000    162.000000
mean    39.703704   30.060494    1.123457  11274.411264
std     14.006161    5.763597    1.259821  10453.660658
min     19.000000   17.400000    0.000000   1727.785000
25%     27.000000   25.900000    0.000000   4895.811250
50%     40.000000   29.750000    1.000000   8530.837000
75%     51.750000   33.650000    2.000000  12642.990500
max     64.000000   47.600000    5.000000  48824.450000

In [119]:
nw_reg_men = nw_reg[(nw_reg["sex"] == "male")]
nw_reg_men.describe()

age         bmi    children       charges
count  163.000000  163.000000  163.000000    163.000000
mean    39.208589   31.129448    1.159509  13412.883576
std     13.952501    5.586314    1.295416  12499.550025
min     19.000000   19.300000    0.000000   1241.565000
25%     26.500000   27.450000    0.000000   4658.699500
50%     39.000000   30.800000    1.000000   9391.346000
75%     51.000000   35.300000    2.000000  17119.976000
max     64.000000   45.900000    5.000000  52590.829390

In [120]:
nw_smoker = nw_reg[(nw_reg["smoker"] == "yes")]
nw_smoker.describe()

age        bmi   children       charges
count  58.000000  58.000000  58.000000     58.000000
mean   36.103448  31.005172   1.206897  32269.063494
std    13.739033   5.420831   1.280880  11270.279593
min    19.000000  18.300000   0.000000  13844.506000
25%    25.000000  27.625000   0.000000  20396.042022
50%    33.500000  31.350000   1.000000  35165.256500
75%    45.500000  34.775000   2.000000  39809.292750
max    64.000000  47.600000   5.000000  52590.829390

In [121]:
nw_bmi = nw_reg[(nw_reg["bmi"] > 30.663397)]
nw_bmi.describe() 

age         bmi    children       charges
count  155.000000  155.000000  155.000000    155.000000
mean    41.683871   35.384516    1.193548  15233.914251
std     14.157167    3.512448    1.238436  14160.770762
min     19.000000   30.800000    0.000000   1261.442000
25%     30.000000   32.800000    0.000000   5568.988500
50%     41.000000   34.800000    1.000000   9872.701000
75%     54.000000   37.100000    2.000000  14338.697685
max     64.000000   47.600000    5.000000  52590.829390

In [122]:
nw_kids = nw_reg[(nw_reg["children"] > 1.094918)]
nw_kids.describe()

age         bmi    children       charges
count  109.000000  109.000000  109.000000    109.000000
mean    39.752294   31.601835    2.688073  14252.615901
std     11.857297    5.484355    0.889207  12250.149406
min     19.000000   18.300000    2.000000   2913.569000
25%     30.000000   27.500000    2.000000   6059.173000
50%     39.000000   30.900000    2.000000   9414.920000
75%     49.000000   34.800000    3.000000  17942.106000
max     64.000000   47.600000    5.000000  47305.305000

The average cost of insurance in this region is $12346.937377
24.29% of the population
47.67% above average bmi
17.85% smoke
50.15% are men
49.85% are women
49.85 are older than the average age
33.54% have more ckildren than average